test/table.ts

```diff
@@ -2014,54 +2045,104 @@ describe('BigQuery/Table', () => {
       }),
     };
 
+    // HACK @types/sinon is missing the timer async methods
+    interface SinonFakeTimersShim extends sinon.SinonFakeTimers {
+      runAllAsync(): Promise<number>;
+      tickAsync(ms: number | string): Promise<number>;
+    }
+
+    let clock: SinonFakeTimersShim;
+    let requestStub: sinon.SinonStub;
+
+    before(() => {
+      clock = sinon.useFakeTimers() as SinonFakeTimersShim;
+    });
+
     beforeEach(() => {
+      requestStub = sinon.stub(table, 'request').resolves([{}]);
       fakeUuid.v4 = () => {
         return fakeInsertId;
       };
     });
 
-    it('should throw an error if rows is empty', () => {
-      assert.throws(() => {
-        table.insert([]);
-      }, /You must provide at least 1 row to be inserted\./);
+    afterEach(() => {
+      clock.reset();
     });
 
-    it('should save data', done => {
-      table.request = (reqOpts: DecorateRequestOptions) => {
-        assert.strictEqual(reqOpts.method, 'POST');
-        assert.strictEqual(reqOpts.uri, '/insertAll');
-        assert.deepStrictEqual(reqOpts.json, dataApiFormat);
-        done();
-      };
-
-      table.insert(data, done);
+    after(() => {
+      clock.restore();
     });
 
-    it('should generate insertId', done => {
-      table.request = (reqOpts: DecorateRequestOptions) => {
-        assert.strictEqual(reqOpts.json.rows[0].insertId, fakeInsertId);
-        done();
-      };
+    /**
+     * Only use this method when NOT directly awaiting on `table.insert`, i.e.
+     * when relying on any of the fake timer async helpers.
+     * Tests should assert isRejected or isFulfilled.
+     * @ignore
+     * @param fn
+     * @returns {Promise<pReflect.PromiseResult<any>>}
+     */
+    async function reflectAfterTimer<FnReturn>(
+      fn: () => Promise<FnReturn>
+    ): Promise<pReflect.PromiseResult<FnReturn>> {
+      /*
```

Comment:
So this is a new one :) Could I trouble you to use a more traditional comment style?  Either:

```js
/**
 * I am a multiline comment rawr
 */
```

Or ...

```js
// this
// is
// fine
// too
```

Reply:
yeah, that's my bad, changed to `//`